src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -150,7 +150,12 @@ private CancellationToken GetCancellationToken(CancellationToken userCancellatio
         private void LogTelemetryData(string region, RegionSource regionSource, RequestContext requestContext)
         {
             requestContext.ApiEvent.RegionDiscovered = region;
-            requestContext.ApiEvent.RegionSource = (int)regionSource;
+
+            if (requestContext.ApiEvent.RegionSource == (int)RegionSource.None)
```

Comment:
>if (requestContext.ApiEvent.RegionSource == (int)RegionSource.None) [](start = 12, length = 67)

why?

Reply:
@neha-bhargava can explain better then me.

Reply:
Since we call this method multiple times in the flow, this value can get overwritten when this method is called the second time and indicate that the value was cached. So, to maintain the original value we have to check if we are not overwriting the value we had.

Reply:
what is the problem of overwriting it? Why would the first value be better than the second value?

Reply:
For example, for a single request the flow goes like following:
1. Region is not discovered yet and env variable is the source of region discovery, region_source will set to 1.
2. The method is called again and in the previous iteration the value gets cached, region_source will set to 3.
This will cause the region_source value to always be 3 which means the region was read from the cache. I am not sure why it was working fine before this fix and the tests and telemetry was showing the correct results.